### PR TITLE
fix: bug, set the default repo visibility to public

### PR DESCRIPTION
**Pull request by davidvader**
fixes a bug when adding a new repo using minimal options

bug to fix:
```bash
$ git checkout master
$ git pull
$ make up
$ vela update config --api.addr http://localhost:8080
$ vela add repo --org DavidVader --repo heyvela
FATA[0000] unable to create new repo DavidVader/heyvela: empty repo visibility provided 
```

(in fact, args to add repo are somewhat ignored. separate bug)

### CODE REVIEW

**Review by wass3r — COMMENTED** [review]
Are you sure? I just tried with `0.6.0` of the CLI after adding a new repository to Github and it worked fine just using `vela add repo --org myorg --repo myrepo`

**Comment by davidvader** [comment]
> Are you sure? I just tried with `0.6.0` of the CLI after adding a new repository to Github and it worked fine just using `vela add repo --org myorg --repo myrepo`

Were you using the unreleased server changes on `master`? This only started occurring after merging with the recent webhookvalidation changes

**Comment by davidvader** [comment]
@wass3r so i was running something like:
```bash
$ git checkout master
$ git pull
$ make up
$ vela update config --api.addr http://localhost:8080
$ vela add repo --org DavidVader --repo heyvela
```

**Comment by wass3r** [comment]
Just wanted to make sure 👍 I did consider the latest commit to check against. Thanks for the confirmation.

**Comment by wass3r** [comment]
I agree, this must be a bigger issue. `GetVisibility()` should be `"public"` by default if not provided in the CLI command (see https://github.com/go-vela/cli/blob/master/action/repo_add.go#L65). I know I've recently been in this part of the code, so it's probably something I changed, though I didn't really touch the part that binds the incoming data to the struct 🤔

File: api/repo.go [hunk 1]
```diff
@@ -112,9 +112,7 @@ func CreateRepo(c *gin.Context) {
 		r.SetTimeout(constants.BuildTimeoutMin)
 	}
 
-	if len(input.GetVisibility()) == 0 {
-		r.SetVisibility(constants.VisibilityPublic)
-	}
+	r.SetVisibility(constants.VisibilityPublic)
```
Comment (jbrockopp) [inline api/repo.go:115]: I think we're close with this change but I think I'd like to see it modified slightly 👍 

The behavior you're experiencing is due to how the code is written.

Previously, we told the API set the `repo.Visibility` field to `public` **if one was not** provided from the `input`.

The problem is we don't do anything with the `repo.Visibility` field **if one was** provided from the `input`.

Instead, I believe the code should look like:

```suggestion
	if len(input.GetVisibility()) == 0 {
		r.SetVisibility(constants.VisibilityPublic)
	} else {
		r.SetVisibility(input.GetVisibility())
	}
```

**Review by jbrockopp — CHANGES_REQUESTED** [review]

File: api/repo.go (hunk 1 above)
Comment (wass3r) [reply api/repo.go:115]: 🤦 looks like i goofed that in a recent change. my bad. good fine @jbrockopp 👍

Comment (davidvader) [reply api/repo.go:115]: yeah youre right, good call, I mentioned that in the PR description and saw it was in updaterepo

**Comment by davidvader** [comment]
changes applied 👍

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by wass3r — APPROVED** [review]

**Pull request merged**